### PR TITLE
feat: update DISCOVERY_VERSION automatically

### PR DESCRIPTION
**Pull request by bruno-fs**
demo here

https://asciinema.org/a/lYhYOaHJ9k6XyMzVJJHoLjdAt

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -175,10 +182,17 @@ def _get_commit_sha(user, repository, commitsh):
     raise typer.Abort()
 
 
-def _side_effects(source, new_commit):
+def _side_effects(source: dict, commitsh: str):
+    """_summary_
```
Comment (infinitewarp) [inline app.py]: Is `_summary_` a placeholder, or does it have some special magic value?

Comment (bruno-fs) [reply app.py]: lol, I forgot to fill it

Comment (bruno-fs) [reply app.py]: it's just a placeholder

File: app.py [hunk 2]
```diff
@@ -197,14 +211,24 @@ def _update_rust_deps_if_required(source, new_commit):
         console.print(f"rust :crab: libraries remain the same ({old_versions}).")
```
Comment (infinitewarp) [inline app.py:217]: LOL… I know this isn't part of your commit, but I was just curious if you know: Where is `:crab:` being translated into the :crab: emoji that I see in your console output?

File: app.py [hunk 3]
```diff
@@ -83,8 +83,8 @@ def update_remote_sources(distgit_path: Path = distgit_path_arg):
             console.print(f"\[{source['name']}] Nothing to update")
         else:
             console.print(f"\[{source['name']}] updating ref to '{commit_sha}'")
-            _side_effects(source, commit_sha)
             source["remote_source"]["ref"] = commit_sha
+            _side_effects(source, commitsh)
```
Comment (infinitewarp) [inline app.py]: Not a major problem, but it took me a minute of looking around to understand the difference here between `commitsh` and `commit_sha`. [The git docs](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish) use "commit-ish" (note the extra `i`) and "committish" (and an extra `t`). Using "committish" might help with the initial visual confusion, but I just wish there was a better name in general. 😕

(Also, I realize you didn't add this variable in this PR. This isn't urgent to fix.)

File: app.py (hunk 2 above)
Comment (bruno-fs) [reply app.py:217]: XD

rich ftw

Comment (bruno-fs) [reply app.py:217]: `python -m rich.markup` and have fun

Comment (bruno-fs) [reply app.py:217]: whoops, i meant `python -m rich.emoji`

In my experience, it seems to be using the same names used on github flavored markdown

File: app.py (hunk 3 above)
Comment (bruno-fs) [reply app.py]: that's where I got the name haha, but wrote it with a typo XD

File: app.py (hunk 2 above)
Comment (infinitewarp) [reply app.py:217]: Nice! _Of course_ it's rich. I ran into this just a few weeks ago and totally forgot!

**Review by infinitewarp — APPROVED** [review]
🚢 🇮🇹 

Also, it tickles something in my brain to see all these variables perfectly aligned. 😍 

```
container_path
committish_map
container_data
perform_update
```

**Pull request merged**